add replacer for JSON unsupported types

matthias-heinisch opened the pull request:
<!--
## Before you hit that Submit button....

Please read our [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
before submitting a pull-request.

## Types of changes

What types of changes does your code introduce?
Put an `x` in the boxes that apply
-->

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

<!--
If you want to raise a pull-request with a new feature, or a refactoring
of existing code, it **may well get rejected** if it hasn't been discussed on
the [forum](https://discourse.nodered.org) or
[slack team](https://nodered.org/slack) first.

-->

## Proposed changes

<!-- Describe the nature of this change. What problem does it address? -->

the cbor parser in nodered will be more robust to JSON unsupported types. see: https://github.com/node-red/node-red-nodes/issues/1101

## Checklist
<!-- Put an `x` in the boxes that apply -->

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass

affected unit tests for cbor are passing. had to ignore the files hardware/Arduino/35-arduino.js and hardware/Arduino/35-arduino.html, other errors occured with FeedParseNode.

- [ ] I have added suitable unit tests to cover the new/changed functionality

matthias-heinisch committed:
add replacer for JSON unsupported types

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

matthias-heinisch committed:
adapt readme and package dependencies

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

alldo-dev commented on parsers/cbor/70-cbor.js:
```diff
@@ -1,49 +1,58 @@
-                        else {
-                            node.status({text:l +" b->o "+ JSON.stringify(value).length});
-                        }
+                        node.status({ text: l + " b->o " + JSON.stringify(value, replacer).length });
```
**nitpick(if-minor)**: Working with strings in javascript is hard as it is. I know it might seem easier to use arithmetic (the plus `+`) to concatenate strings but it can potentially lead to a bunch of problems.

Specially because you are concatenating the value of `l` (which results in an `int` type from the line 32 `var l = value.length;` then another string value (`" b->o "`) then what seems to be a string turned to number. 

IMHO it would be cleaner to turn everything into a string (parsing the values first) and using template literals.

```javascript
var valueLengthAsString = value.length.toString();
var replacedValuesLengthAsString = JSON.stringify(value, replacer).length
var textValue = `${valueLengthAsString} b->o ${replacedValuesLengthAsString}`
node.status({ text: textValue });
```

alldo-dev commented on parsers/cbor/70-cbor.js:
```diff
@@ -1,49 +1,58 @@
+                    RED.util.setMessageProperty(msg, node.property, value);
                     node.send(msg);
-                    node.status({text:le +" o->b "+ value.length});
+                    node.status({ text: le + " o->b " + value.length });
```
**nit-pick(if-minor)**: Same case as before

matthias-heinisch committed:
Enhancing status string robustness

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

alldo-dev committed:
Update 70-cbor.js

fixed typo in variable names, when using template literals the values are parsed to Strings. Hence, renamed to `*LengthAsString` to simply `*Length`
